assets/src/dashboard/components/toaster/test/useToastContext.js

```diff
@@ -0,0 +1,155 @@
+
+/**
+ * Internal dependencies
+ */
+import ToastProvider from '../provider';
+import useToastContext from '../useToastContext';
+import { ALERT_SEVERITY } from '../../../constants';
+
+const TOAST_1 = {
+  message: 'one',
+  severity: ALERT_SEVERITY.ERROR,
+  id: 1,
+  isActive: true,
+};
+
+const TOAST_2 = {
+  message: 'two',
+  id: 2,
+  severity: ALERT_SEVERITY.ERROR,
+  isActive: true,
+};
+
+const TOAST_3 = {
+  message: 'three',
+  id: 3,
+  severity: ALERT_SEVERITY.WARNING,
+  isActive: true,
+};
+
+describe('useToastContext', () => {
+  it('should throw an error if used outside of Toast.Provider', () => {
+    expect(() => {
+      const {
+        // eslint-disable-next-line no-unused-vars
+        result: { current },
+      } = renderHook(() => useToastContext());
+    }).toThrow(
+      Error('useToasterContext() must be used within a <Toast.Provider />')
+    );
+  });
+
+  it('should not throw an error if used inside ToastProvider', () => {
+    const { result } = renderHook(() => useToastContext(), {
+      wrapper: ToastProvider,
+    });
+    expect(result.current.error).toBeUndefined();
+  });
+
+  it('should have default state initially set up', () => {
+    const { result } = renderHook(() => useToastContext(), {
+      wrapper: ToastProvider,
+    });
+
+    expect(result.current.state.activeToasts).toStrictEqual([]);
+  });
+
+  it('should add a new activeToast when addToast is called and new toast has unique id', () => {
+    const { result } = renderHook(() => useToastContext(), {
+      wrapper: ToastProvider,
+    });
+
+    act(() => {
+      result.current.actions.addToast(TOAST_1);
+    });
+
+    expect(result.current.state.activeToasts).toStrictEqual([TOAST_1]);
+  });
+
+  it('should not add a duplicate activeToast when addToast is called with existing toast id', () => {
+    const { result } = renderHook(() => useToastContext(), {
+      wrapper: ToastProvider,
+    });
+    act(() => {
+      result.current.actions.addToast(TOAST_1);
+    });
+
+    act(() => {
+      result.current.actions.addToast(TOAST_1);
+    });
+
+    expect(result.current.state.activeToasts).toStrictEqual([TOAST_1]);
+  });
+
+  it('should remove an activeToast but maintain allToasts when removeToast is called', () => {
+    const { result } = renderHook(() => useToastContext(), {
+      wrapper: ToastProvider,
+    });
+
+    act(() => {
+      result.current.actions.addToast(TOAST_1);
+    });
+    expect(result.current.state.activeToasts).toStrictEqual([TOAST_1]);
+
+    act(() => {
+      result.current.actions.addToast(TOAST_2);
+    });
+    expect(result.current.state.activeToasts).toStrictEqual([TOAST_1, TOAST_2]);
+
+    act(() => {
+      result.current.actions.removeToast(TOAST_2.id);
+    });
+
+    expect(result.current.state.activeToasts).toStrictEqual([TOAST_1]);
+
+    act(() => {
+      result.current.actions.removeToast(TOAST_1.id);
+    });
+    expect(result.current.state.activeToasts).toStrictEqual([]);
+  });
```

Comment:
this is kind of a nit, but how are we testing the `but maintain allToasts when removeToast is called` portion of this test? seems like maybe it could be removed from the description of the test 🤷